.github/workflows/pull-request.yml

```diff
@@ -195,14 +195,16 @@ jobs:
         id: changed-files
         uses: tj-actions/changed-files@v45
 
-      - name: Set up Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: 3.9
+      - name: Filter TypeScript files
+        id: filter-files
+        run: |
+          echo "ts_files=$(echo '${{ steps.changed-files.outputs.all_changed_files }}' | tr ',' '\n' | grep -E '\.(ts|tsx)$' | tr '\n' ' ')" >> $GITHUB_OUTPUT
```

Comment:
1. This isn't necessary. The script evaluates the type of file by its extension. Only the changed files in the PR should be used. 
2. Normally there are so few files changed that this optimization isn't required.
3. Please revert.